src/util/DeviceFingerprint.js

```diff
@@ -24,6 +28,12 @@ define(['vendor/lib/q', 'okta/jquery'], function (Q, $) {
       });
       $iframe.appendTo(element);
 
+      function isWindowsPhone() {
+        return navigator.userAgent.match(/Windows Phone/i) ||
```

Comment:
just to be safe, can we add a check for navigator.userAgent not being undefined? (I can't think of how this would happen, but maybe there is a bad embedded browser that can screw this up)